samples/js/photos/src/photo.js

```diff
@@ -8,20 +8,21 @@ import type {NomsPhoto, PhotoSize} from './types.js';
 
 import type {MapEntry} from '@attic/noms';
 import {
+  Blob,
   StructMirror,
   equals,
   invariant,
   notNull,
 } from '@attic/noms';
-export type SizeEntry = MapEntry<PhotoSize, string>;
+export type SizeEntry = MapEntry<PhotoSize, Blob>;
 
 /**
  * createPhoto asynchronously derives a Photo instance from a NomsPhoto.
  */
 export function createPhoto(path: string, p: NomsPhoto): Promise<Photo> {
   const sizeEntries = [];
-  return p.sizes.forEach((url, size) => {
-    sizeEntries.push([size, url]);
+  return p.resources.forEach((details, size) => {
```

Comment:
For the sake of the mobile UI, we should probably have both the sizes and the resources on the photo.